pandas/__init__.py

```diff
@@ -297,32 +297,71 @@ def __getattr__(self, item):
 
     np = __numpy()
 
-    class __Datetime:
-        def __init__(self):
-            from datetime import datetime as dt
+    class __Datetime(type):
 
-            self.datetime = dt
+        from datetime import datetime as dt
 
-        def __getattr__(self, item):
+        datetime = dt
+
+        def __getattr__(cls, item):
+            cls.emit_warning()
+
+            try:
+                return getattr(cls.datetime, item)
+            except AttributeError:
+                raise AttributeError(f"module datetime has no attribute {item}")
+
+        def __instancecheck__(cls, other):
+            return isinstance(other, cls.datetime)
+
+    class __DatetimeSub(metaclass=__Datetime):
+        def emit_warning(dummy=0):
             import warnings
 
             warnings.warn(
                 "The pandas.datetime class is deprecated "
                 "and will be removed from pandas in a future version. "
                 "Import from datetime instead.",
                 FutureWarning,
-                stacklevel=2,
+                stacklevel=3,
             )
 
-            try:
-                return getattr(self.datetime, item)
-            except AttributeError:
-                raise AttributeError(f"module datetime has no attribute {item}")
+        def __new__(cls, *args, **kwargs):
+            cls.emit_warning()
+            from datetime import datetime as dt
 
-    datetime = __Datetime().datetime
+            return dt(*args, **kwargs)
 
-    class SparseArray:
-        pass
+    datetime = __DatetimeSub
+
+    class __SparseArray(type):
+
+        from pandas.core.arrays.sparse import SparseArray as sa
+
+        SparseArray = sa
```

Comment:
is this needed? if you ```from pandas.core.arrays.sparse import SparseArray``` does it still work?

Reply:
You're right.  I was following the pattern used for `datetime`, but that was unnecessary, so I changed it there and in `SparseArray`